pandas-stubs/core/indexes/datetimes.pyi

```diff
@@ -97,6 +97,7 @@ class DatetimeIndex(  # type: ignore[misc]
     def tzinfo(self) -> tzinfo | None: ...
     @property
     def dtype(self) -> np.dtype | DatetimeTZDtype: ...
+    def intersection(self, other: DatetimeIndex | list) -> DatetimeIndex: ...
```

Comment:
`list` is too broad.  I think `Sequence[DatetimeLike]` will work where you import `DatetimeLike` from `pandas._typing`